bin/iex

```diff
@@ -43,6 +43,14 @@ readlink_f () {
   fi
 }
 
+ERL="-noshell -user Elixir.IEx.CLI"
 SELF=$(readlink_f "$0")
 SCRIPT_PATH=$(dirname "$SELF")
-exec "$SCRIPT_PATH"/elixir --no-halt --erl "-noshell -user Elixir.IEx.CLI" +iex "$@"
+
+if [ -z "$HOME" ]; then
+  :
+else
+  ERL="-kernel shell_history_path \"$HOME/.elixir-history\" $ERL"
+fi
```

Comment:
@josevalim I'll do it but can't promise it for today or Friday -- might be the weekend. I'll account for these cases in a PR:

* Directory without spaces
* Directory with spaces (you're right that quoting might be necessary)
* Directory which is an NTFS symbolic link
* Directory which is an NTFS junction point (ref: https://stackoverflow.com/questions/9042542/what-is-the-difference-between-ntfs-junction-points-and-symbolic-links)

It's definitely gonna be a bit larger than the straightforward script above, but it should be quite a bit more bulletproof.